src/lib/lore.sh

```diff
@@ -458,3 +474,169 @@ function download_series()
   done
   wait
 }
+
+# This function deletes a patch series from the local storage
+#
+# @download_dir_path: The path to the directory where the series was stored
+# @patch_title: The title of the patch
+# @flag: Flag to control function output
+#
+# Return:
+# Return 0 if there are files related to the patch to be deleted and 2 (ENOENT)
+# if there aren't.
+function delete_series_from_local_storage()
+{
+  local download_dir_path="$1"
+  local patch_title="$2"
+  local flag="$3"
+  local filename_pattern
+  local files_to_remove
+  local cmd
+  local ret
+
+  flag=${flag:-'SILENT'}
+
+  filename_pattern=$(convert_title_to_patch_name "$patch_title")
+
+  # Find doesn't return a non-zero status code when not matching any file.
+  # We use grep to work-around it.
+  files_to_remove=$(find "${download_dir_path}" -iname "*${filename_pattern}" | grep '.')
+  ret="$?"
+
+  if [[ "$ret" == 0 ]]; then
+    while IFS=' ' read -r file; do
+      # Just to avoid trying to remove fs root
+      if [[ "$file" != '/' ]]; then
+        cmd="rm ${file}"
+        cmd_manager "$flag" "$cmd"
+      fi
+    done <<< "${files_to_remove}"
+  else
+    return 2 # ENOENT
+  fi
+}
+
+# This function creates the lore bookmarked series file if it doesn't
+# already exists
+function create_lore_bookmarked_file()
+{
+  create_lore_data_dir
+
+  [[ -f "${BOOKMARKED_SERIES_PATH}" ]] && return
+  touch "${BOOKMARKED_SERIES_PATH}"
+}
+
+# This function adds a given series to a local bookmark database for
+# series managed by kw. To achieve uniqueness for each series the function
+# assigns an ID and timestamp (for informational purposes).
+#
+# @target_patch: Patch (series) in the same format as list_of_mailinglist_patches
+#   to be added to local bookmark database
+# @download_dir_path: The directory where the patch (series) was saved
+function add_series_to_bookmark()
+{
+  local target_patch="$1"
+  local download_dir_path="$2"
+  local patch_id
+  local timestamp
+  local count
+
+  create_lore_bookmarked_file
+
+  patch_id=$(printf '%s' "${target_patch}" | sha256sum | cut -d ' ' -f1)
```

Comment:
Don't we need to add any specific package in the setup phase to ensure that sha256sum is available?